File: benchmark/prime.js
#!/usr/bin/env nodejs

function prime(n)
{
    var table = [ false, false ], count = 0;
    for (var i = 2; i < n; i++) {
        table.push(true);
    }
    for (var i = 2; i < n; i++) {
        if (table[i]) {
            count++;
            for (var j = i * 2; j < n; j += i) {
                table[j] = false;
            }
        }
    }
    return count;
}

process.stdin.once('data', function (val){
    var n = parseInt(val, 10);
    var primes = prime(n);
    console.log(primes);
}).resume();

File: benchmark/prime.py
#!/usr/bin/env python3

def prime(n):
    table = [ False, False ] + [ True ] * (n - 2)
    count = 0

    for v, is_prime in enumerate(table):
        if is_prime:
            count += 1
            for j in range(v * 2, n, v):
                table[j] = False

    return count

primes = prime(int(input()))
print(primes)
